File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/Rate.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Entity
@Table(name = "rate")
public class Rate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "r_id")
    private Long rid;

    @Column(name = "r_value", nullable = false, columnDefinition = "DECIMAL(2,1)")
    private BigDecimal rvalue;

    @Column(name = "r_date", nullable = false)
    private java.sql.Timestamp rdate;

    @Column(name = "r_comment", length = 5120)
    private String rcomment;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "m_id", nullable = false)
    private Long mid;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;
}

File: Frontend/manga-world/src/Model/Rate.ts
import Manga from "./Manga";
import User from "./User";

class Rate {
    rId: number;
    rValue: number;
    rDate: Date;
    rComment: string;
    userId: number;
    mId: number;
    user: User;
    manga: Manga;
    
    constructor(rId: number, rValue: number, rDate: Date, rComment: string, userId: number, mId: number, user: User, manga: Manga) {
        this.rId = rId;
        this.rValue = rValue;
        this.rDate = rDate;
        this.rComment = rComment;
        this.userId = userId;
        this.mId = mId;
        this.user = user;
        this.manga = manga;
    }
}

export default Rate;